### PR TITLE
Add fonts for new builtin Shiny theme preset

### LINKED ISSUE #630: Custom font settings for shiny theme
Use custom font settings for the shiny theme using a sass file and bundled fonts rather than via the `bs_theme()` function. This will ensure that the font settings can be overridden or removed from the theme bundle via `bs_theme_update()`.

### PR DESCRIPTION
**Pull request by gadenbuie**
Closes #630 and adds the desired font stack for the new Shiny theme, namely Open Sans and Source Code Pro.

This PR updates the `tools/download_bootswatch_fonts.R` script to also download fonts for builtin presets. This necessitated a few changes:

1. `tools/download_bootswatch_fonts.R` is now `tools/download_preset_fonts.R`
2. The download fonts script now looks for `_bootswatch.scss` or `_rules.scss` files (in that order)
3. Builtin presets now must follow this convention:
    * Preset files are stored in `inst/bs{version}/{name}/`
    * Preset scss files should be named according to their layer, e.g. `_variables.scss` and `_rules.scss`.

From what I can tell, these new fonts add ~600KB to the package bundle. We may want to cut back on font weights or figure out how to reduce the package bundle size, but I'm also okay waiting for on that for another PR.

### CODE REVIEW

File: inst/builtin/bs5/shiny/_rules.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*-- scss:rules --*/
+$web-font-path: "https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,300;0,400;0,500;0,600;0,700;0,800;1,300;1,400;1,500;1,600;1,700;1,800&family=Source+Code+Pro:ital,wght@0,400;0,600;1,400;1,600&display=swap" !default;
```
Comment (gadenbuie) [inline inst/builtin/bs5/shiny/_rules.scss:2]: @gregswinehart Do you think we could cut back these font weights a little bit?

File: tools/download_preset_fonts.R [hunk 2]
```diff
@@ -15,7 +15,16 @@ extract_first_group <- function(x, pattern) {
 }
 
 download_and_copy_fonts <-  function(theme) {
-  theme_scss <- readLines(file.path(theme, "_bootswatch.scss"))
+  theme_file_choices <- paste0(c("_bootswatch", "_rules"), ".scss")
+  theme_files <- file.path(theme, theme_file_choices)
```
Comment (schloerke) [inline tools/download_preset_fonts.R:19]: Can the file choice be given as a variable?

```suggestion
download_and_copy_fonts <-  function(theme, rule_file = "_rules.scss") {
  theme_files <- file.path(theme, rule_file)
```

File: tools/download_preset_fonts.R [hunk 3]
```diff
@@ -65,7 +74,14 @@ themes3 <- list.dirs(
...
+  "inst/builtin/bs5",
+  recursive = FALSE,
+  full.names = TRUE
+)
 
 lapply(themes5, download_and_copy_fonts)
 lapply(themes4, download_and_copy_fonts)
 lapply(themes3, download_and_copy_fonts)
+
+lapply(builtin5, download_and_copy_fonts)
```
Comment (schloerke) [inline tools/download_preset_fonts.R:87]:
```suggestion
lapply(themes5, download_and_copy_fonts, "_bootswatch.scss")
lapply(themes4, download_and_copy_fonts, "_bootswatch.scss")
lapply(themes3, download_and_copy_fonts, "_bootswatch.scss")

lapply(builtin5, download_and_copy_fonts, "_rules.scss")
```

**Review by schloerke — APPROVED** [review]

File: inst/builtin/bs5/shiny/_rules.scss (hunk 1 above)
Comment (gregswinehart) [reply inst/builtin/bs5/shiny/_rules.scss:2]: I suppose we could if it's a thing. That said, I would probably only lose the 800 weight? I like to celebrate the whole catalog....

File: tools/download_preset_fonts.R (hunk 2 above)
Comment (gadenbuie) [reply tools/download_preset_fonts.R]: Good call, thanks! Done in cfd03d29b6f06d14c4f5f69f1de10da31944b479

File: inst/builtin/bs5/shiny/_rules.scss (hunk 1 above)
Comment (gadenbuie) [reply inst/builtin/bs5/shiny/_rules.scss:2]: Thanks Greg. I'm going to leave this as is for now and will review again with Carson this week.

**Pull request merged**